### DISCUSSION TITLE
save_options quesion in save_options.cpp

### DISCUSSION THREAD

**Discussion by Mr-zhangyh** [zeux/pugixml#645] [Q&A] [answered]
When I output an XML file, how do I put the XML declaration on the first line and the other root nodes on the second line?
I didn't find an example in the sample code save_options.cpp.
eg：
<?xml version="1.0" encoding="utf-8" standalone="yes"?>
<root><child>txt</child></root>

**Comment by zeux** [comment, marked as answer]
You would need to use `pugi::format_raw` and insert the declaration (if necessary) as well as the required newlines yourself, for example:

```
    pugi::xml_node decl = doc.prepend_child(pugi::node_declaration);
    decl.append_attribute("version").set_value("1.0");
    decl.append_attribute("encoding").set_value("utf-8");
    decl.append_attribute("standalone").set_value("yes");

    doc.insert_child_after(pugi::node_pcdata, decl).set_value("\n");
    doc.append_child(pugi::node_pcdata).set_value("\n");

    doc.save(std::cout, "", pugi::format_raw);
```
